### PR TITLE
GUM-66: Fixed date input error styles and validation bugs

### PR DESCRIPTION
**Pull request by katyakostikova**

### CODE REVIEW

File: src/components/form/form-date/form-date.tsx [hunk 1]
```diff
@@ -28,17 +35,30 @@ const FormDate: React.FC<FormDateProps> = ({
   const error = touched[name] && errors[name];
 
   const handleChange = (date: Date | undefined) => setFieldValue(name, date);
+  const handleManualEdit = (
```
Comment (R-Mielamud) [inline src/components/form/form-date/form-date.tsx]: Add a linebreak between functions

**Review by R-Mielamud — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

File: src/components/form/form-date/form-date.tsx [hunk 2]
```diff
@@ -29,16 +36,30 @@ const FormDate: React.FC<FormDateProps> = ({
 
   const handleChange = (date: Date | undefined) => setFieldValue(name, date);
 
+  const handleManualEdit = (
+    e: NativeSyntheticEvent<TextInputEndEditingEventData>
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-date/form-date.tsx:40]:
```suggestion
    { nativeEvent }: NativeSyntheticEvent<TextInputEndEditingEventData>
```

File: src/components/form/form-date/form-date.tsx [hunk 3]
```diff
@@ -29,16 +36,30 @@ const FormDate: React.FC<FormDateProps> = ({
...
+  ) => {
+    const { text } = e.nativeEvent;
+    const date = dayjs(text);
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-date/form-date.tsx:43]: Always pass parsing format as second param.

File: src/components/form/form-date/form-date.tsx [hunk 4]
```diff
@@ -29,16 +36,30 @@ const FormDate: React.FC<FormDateProps> = ({
...
+    if (date.isValid() && date.year() > 1800) {
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-date/form-date.tsx:44]: Please take 1800 to a constant and... Do you think we should allow immortals to use our app?

File: src/components/form/form-date/form-date.tsx [hunk 5]
```diff
@@ -29,16 +36,30 @@ const FormDate: React.FC<FormDateProps> = ({
...
+      setFieldValue(name, new Date(text));
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-date/form-date.tsx:45]:
```suggestion
      setFieldValue(name, date.toDate());
```

**Review by AleksandrTermenzhy — APPROVED** [review]

**Pull request merged**